packages/plugins/extended-validation/src/rules/one-of.ts

```diff
@@ -38,22 +38,54 @@ export const OneOfInputObjectsRule: ExtendedValidationRule = (validationContext,
           const argType = fieldType.args.find(typeArg => typeArg.name === arg.name.value);
 
           if (argType) {
-            const inputType = unwrapType(argType.type);
-            const isOneOfInputType =
-              inputType.extensions?.oneOf || (inputType.astNode && getDirectiveFromAstNode(inputType.astNode, 'oneOf'));
-
-            if (isOneOfInputType) {
-              const argValue = values[arg.name.value] || {};
-
-              if (Object.keys(argValue).length !== 1) {
-                validationContext.reportError(
-                  new GraphQLError(`Exactly one key must be specified for input type "${inputType.name}"`, [arg])
-                );
-              }
-            }
+            traverseVariables(validationContext, arg, argType.type, values[arg.name.value]);
           }
         }
       }
     },
   };
 };
+
+function traverseVariables(
+  validationContext: ValidationContext,
+  arg: ArgumentNode,
+  graphqlType: GraphQLInputType,
+  currentValue: unknown
+) {
+  // if the current value is empty we don't need to traverse deeper
+  // if it shouldn't be empty, the "original" validation phase should complain.
+  if (currentValue == null) {
+    return;
+  }
+
+  if (isListType(graphqlType)) {
+    // if it is a list type currentValue MUST be an array
+    assertArray(currentValue);
```

Comment:
I think we should not assert here and instead skip the validation if currentValue is not an array. The original validation should complain.